vortex-duckdb/src/convert/array/mod.rs

```diff
@@ -33,6 +36,7 @@ use crate::{DUCKDB_STANDARD_VECTOR_SIZE, ToDuckDBType};
 #[derive(Default)]
 pub struct ConversionCache {
     pub values_cache: HashMap<usize, FlatVector>,
+    pub canonical_cache: HashMap<usize, Canonical>,
```

Comment:
what's the lifetime of the `ConvesionCache`?

Reply:
along those lines, should we limit the number of entries, have some cache eviction policy?

Reply:
The scope of the chunks returned by vortex after splitting

Reply:
if `ConvesionCache` lives a long time, it can consume a ton of memory for data no one is reading